// googleapis/build/src/index.js
"use strict";
// Copyright 2020 Google LLC
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
/*! THIS FILE IS AUTO-GENERATED */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleApis = exports.google = void 0;
const googleapis_1 = require("./googleapis");
Object.defineProperty(exports, "GoogleApis", { enumerable: true, get: function () { return googleapis_1.GoogleApis; } });
const google = new googleapis_1.GoogleApis();
exports.google = google;
exports.Common = require("googleapis-common");
exports.Auth = require("google-auth-library");
var v1_1 = require("./apis/abusiveexperiencereport/v1");
Object.defineProperty(exports, "abusiveexperiencereport_v1", { enumerable: true, get: function () { return v1_1.abusiveexperiencereport_v1; } });
var v1_2 = require("./apis/acceleratedmobilepageurl/v1");
Object.defineProperty(exports, "acceleratedmobilepageurl_v1", { enumerable: true, get: function () { return v1_2.acceleratedmobilepageurl_v1; } });
var v1_3 = require("./apis/accessapproval/v1");
Object.defineProperty(exports, "accessapproval_v1", { enumerable: true, get: function () { return v1_3.accessapproval_v1; } });
var v1beta1_1 = require("./apis/accessapproval/v1beta1");
Object.defineProperty(exports, "accessapproval_v1beta1", { enumerable: true, get: function () { return v1beta1_1.accessapproval_v1beta1; } });
var v1_4 = require("./apis/accesscontextmanager/v1");
Object.defineProperty(exports, "accesscontextmanager_v1", { enumerable: true, get: function () { return v1_4.accesscontextmanager_v1; } });
var v1beta_1 = require("./apis/accesscontextmanager/v1beta");
Object.defineProperty(exports, "accesscontextmanager_v1beta", { enumerable: true, get: function () { return v1beta_1.accesscontextmanager_v1beta; } });
var v1_2_1 = require("./apis/adexchangebuyer/v1.2");
Object.defineProperty(exports, "adexchangebuyer_v1_2", { enumerable: true, get: function () { return v1_2_1.adexchangebuyer_v1_2; } });
var v1_3_1 = require("./apis/adexchangebuyer/v1.3");
Object.defineProperty(exports, "adexchangebuyer_v1_3", { enumerable: true, get: function () { return v1_3_1.adexchangebuyer_v1_3; } });
var v1_4_1 = require("./apis/adexchangebuyer/v1.4");
Object.defineProperty(exports, "adexchangebuyer_v1_4", { enumerable: true, get: function () { return v1_4_1.adexchangebuyer_v1_4; } });
var v2beta1_1 = require("./apis/adexchangebuyer2/v2beta1");
Object.defineProperty(exports, "adexchangebuyer2_v2beta1", { enumerable: true, get: function () { return v2beta1_1.adexchangebuyer2_v2beta1; } });
var v1_5 = require("./apis/adexperiencereport/v1");
Object.defineProperty(exports, "adexperiencereport_v1", { enumerable: true, get: function () { return v1_5.adexperiencereport_v1; } });
var datatransfer_v1_1 = require("./apis/admin/datatransfer_v1");
Object.defineProperty(exports, "admin_datatransfer_v1", { enumerable: true, get: function () { return datatransfer_v1_1.admin_datatransfer_v1; } });
var directory_v1_1 = require("./apis/admin/directory_v1");
Object.defineProperty(exports, "admin_directory_v1", { enumerable: true, get: function () { return directory_v1_1.admin_directory_v1; } });
var reports_v1_1 = require("./apis/admin/reports_v1");
Object.defineProperty(exports, "admin_reports_v1", { enumerable: true, get: function () { return reports_v1_1.admin_reports_v1; } });
var v1_6 = require("./apis/admob/v1");
Object.defineProperty(exports, "admob_v1", { enumerable: true, get: function () { return v1_6.admob_v1; } });
var v1_4_2 = require("./apis/adsense/v1.4");
Object.defineProperty(exports, "adsense_v1_4", { enumerable: true, get: function () { return v1_4_2.adsense_v1_4; } });
var v4_1_1 = require("./apis/adsensehost/v4.1");
Object.defineProperty(exports, "adsensehost_v4_1", { enumerable: true, get: function () { return v4_1_1.adsensehost_v4_1; } });
var v1beta1_2 = require("./apis/alertcenter/v1beta1");
Object.defineProperty(exports, "alertcenter_v1beta1", { enumerable: true, get: function () { return v1beta1_2.alertcenter_v1beta1; } });
var v3_1 = require("./apis/analytics/v3");
Object.defineProperty(exports, "analytics_v3", { enumerable: true, get: function () { return v3_1.analytics_v3; } });
var v1alpha_1 = require("./apis/analyticsadmin/v1alpha");
Object.defineProperty(exports, "analyticsadmin_v1alpha", { enumerable: true, get: function () { return v1alpha_1.analyticsadmin_v1alpha; } });
var v1alpha_2 = require("./apis/analyticsdata/v1alpha");
Object.defineProperty(exports, "analyticsdata_v1alpha", { enumerable: true, get: function () { return v1alpha_2.analyticsdata_v1alpha; } });
var v4_1 = require("./apis/analyticsreporting/v4");
Object.defineProperty(exports, "analyticsreporting_v4", { enumerable: true, get: function () { return v4_1.analyticsreporting_v4; } });
var v1_7 = require("./apis/androiddeviceprovisioning/v1");
Object.defineProperty(exports, "androiddeviceprovisioning_v1", { enumerable: true, get: function () { return v1_7.androiddeviceprovisioning_v1; } });
var v1_8 = require("./apis/androidenterprise/v1");
Object.defineProperty(exports, "androidenterprise_v1", { enumerable: true, get: function () { return v1_8.androidenterprise_v1; } });
var v1_9 = require("./apis/androidmanagement/v1");
Object.defineProperty(exports, "androidmanagement_v1", { enumerable: true, get: function () { return v1_9.androidmanagement_v1; } });
var v1_1_1 = require("./apis/androidpublisher/v1.1");
Object.defineProperty(exports, "androidpublisher_v1_1", { enumerable: true, get: function () { return v1_1_1.androidpublisher_v1_1; } });
var v1_10 = require("./apis/androidpublisher/v1");
Object.defineProperty(exports, "androidpublisher_v1", { enumerable: true, get: function () { return v1_10.androidpublisher_v1; } });
var v2_1 = require("./apis/androidpublisher/v2");
Object.defineProperty(exports, "androidpublisher_v2", { enumerable: true, get: function () { return v2_1.androidpublisher_v2; } });
var v3_2 = require("./apis/androidpublisher/v3");
Object.defineProperty(exports, "androidpublisher_v3", { enumerable: true, get: function () { return v3_2.androidpublisher_v3; } });
var v1beta_2 = require("./apis/apigateway/v1beta");
Object.defineProperty(exports, "apigateway_v1beta", { enumerable: true, get: function () { return v1beta_2.apigateway_v1beta; } });
var v1_11 = require("./apis/appengine/v1");
Object.defineProperty(exports, "appengine_v1", { enumerable: true, get: function () { return v1_11.appengine_v1; } });
var v1alpha_3 = require("./apis/appengine/v1alpha");
Object.defineProperty(exports, "appengine_v1alpha", { enumerable: true, get: function () { return v1alpha_3.appengine_v1alpha; } });
var v1beta_3 = require("./apis/appengine/v1beta");
Object.defineProperty(exports, "appengine_v1beta", { enumerable: true, get: function () { return v1beta_3.appengine_v1beta; } });
var v1_12 = require("./apis/appsactivity/v1");
Object.defineProperty(exports, "appsactivity_v1", { enumerable: true, get: function () { return v1_12.appsactivity_v1; } });
var v1alpha1_1 = require("./apis/area120tables/v1alpha1");
Object.defineProperty(exports, "area120tables_v1alpha1", { enumerable: true, get: function () { return v1alpha1_1.area120tables_v1alpha1; } });
var v1beta1_3 = require("./apis/artifactregistry/v1beta1");
Object.defineProperty(exports, "artifactregistry_v1beta1", { enumerable: true, get: function () { return v1beta1_3.artifactregistry_v1beta1; } });
var v1beta1_4 = require("./apis/assuredworkloads/v1beta1");
Object.defineProperty(exports, "assuredworkloads_v1beta1", { enumerable: true, get: function () { return v1beta1_4.assuredworkloads_v1beta1; } });
var v2_2 = require("./apis/bigquery/v2");
Object.defineProperty(exports, "bigquery_v2", { enumerable: true, get: function () { return v2_2.bigquery_v2; } });
var v1beta1_5 = require("./apis/bigqueryconnection/v1beta1");
Object.defineProperty(exports, "bigqueryconnection_v1beta1", { enumerable: true, get: function () { return v1beta1_5.bigqueryconnection_v1beta1; } });
var v1_13 = require("./apis/bigquerydatatransfer/v1");
Object.defineProperty(exports, "bigquerydatatransfer_v1", { enumerable: true, get: function () { return v1_13.bigquerydatatransfer_v1; } });
var v1_14 = require("./apis/bigqueryreservation/v1");
Object.defineProperty(exports, "bigqueryreservation_v1", { enumerable: true, get: function () { return v1_14.bigqueryreservation_v1; } });
var v1alpha2_1 = require("./apis/bigqueryreservation/v1alpha2");
Object.defineProperty(exports, "bigqueryreservation_v1alpha2", { enumerable: true, get: function () { return v1alpha2_1.bigqueryreservation_v1alpha2; } });
var v1beta1_6 = require("./apis/bigqueryreservation/v1beta1");
Object.defineProperty(exports, "bigqueryreservation_v1beta1", { enumerable: true, get: function () { return v1beta1_6.bigqueryreservation_v1beta1; } });
var v1_15 = require("./apis/bigtableadmin/v1");
Object.defineProperty(exports, "bigtableadmin_v1", { enumerable: true, get: function () { return v1_15.bigtableadmin_v1; } });
var v2_3 = require("./apis/bigtableadmin/v2");
Object.defineProperty(exports, "bigtableadmin_v2", { enumerable: true, get: function () { return v2_3.bigtableadmin_v2; } });
var v1_16 = require("./apis/billingbudgets/v1");
Object.defineProperty(exports, "billingbudgets_v1", { enumerable: true, get: function () { return v1_16.billingbudgets_v1; } });
var v1beta1_7 = require("./apis/billingbudgets/v1beta1");
Object.defineProperty(exports, "billingbudgets_v1beta1", { enumerable: true, get: function () { return v1beta1_7.billingbudgets_v1beta1; } });
var v1_17 = require("./apis/binaryauthorization/v1");
Object.defineProperty(exports, "binaryauthorization_v1", { enumerable: true, get: function () { return v1_17.binaryauthorization_v1; } });
var v1beta1_8 = require("./apis/binaryauthorization/v1beta1");
Object.defineProperty(exports, "binaryauthorization_v1beta1", { enumerable: true, get: function () { return v1beta1_8.binaryauthorization_v1beta1; } });
var v2_4 = require("./apis/blogger/v2");
Object.defineProperty(exports, "blogger_v2", { enumerable: true, get: function () { return v2_4.blogger_v2; } });
var v3_3 = require("./apis/blogger/v3");
Object.defineProperty(exports, "blogger_v3", { enumerable: true, get: function () { return v3_3.blogger_v3; } });
var v1_18 = require("./apis/books/v1");
Object.defineProperty(exports, "books_v1", { enumerable: true, get: function () { return v1_18.books_v1; } });
var v3_4 = require("./apis/calendar/v3");
Object.defineProperty(exports, "calendar_v3", { enumerable: true, get: function () { return v3_4.calendar_v3; } });
var v1_19 = require("./apis/chat/v1");
Object.defineProperty(exports, "chat_v1", { enumerable: true, get: function () { return v1_19.chat_v1; } });
var v1_20 = require("./apis/chromeuxreport/v1");
Object.defineProperty(exports, "chromeuxreport_v1", { enumerable: true, get: function () { return v1_20.chromeuxreport_v1; } });
var v2_5 = require("./apis/civicinfo/v2");
Object.defineProperty(exports, "civicinfo_v2", { enumerable: true, get: function () { return v2_5.civicinfo_v2; } });
var v1_21 = require("./apis/classroom/v1");
Object.defineProperty(exports, "classroom_v1", { enumerable: true, get: function () { return v1_21.classroom_v1; } });
var v1_22 = require("./apis/cloudasset/v1");
Object.defineProperty(exports, "cloudasset_v1", { enumerable: true, get: function () { return v1_22.cloudasset_v1; } });
var v1beta1_9 = require("./apis/cloudasset/v1beta1");
Object.defineProperty(exports, "cloudasset_v1beta1", { enumerable: true, get: function () { return v1beta1_9.cloudasset_v1beta1; } });
var v1p1beta1_1 = require("./apis/cloudasset/v1p1beta1");
Object.defineProperty(exports, "cloudasset_v1p1beta1", { enumerable: true, get: function () { return v1p1beta1_1.cloudasset_v1p1beta1; } });
var v1p4beta1_1 = require("./apis/cloudasset/v1p4beta1");
Object.defineProperty(exports, "cloudasset_v1p4beta1", { enumerable: true, get: function () { return v1p4beta1_1.cloudasset_v1p4beta1; } });
var v1p5beta1_1 = require("./apis/cloudasset/v1p5beta1");
Object.defineProperty(exports, "cloudasset_v1p5beta1", { enumerable: true, get: function () { return v1p5beta1_1.cloudasset_v1p5beta1; } });
var v1_23 = require("./apis/cloudbilling/v1");
Object.defineProperty(exports, "cloudbilling_v1", { enumerable: true, get: function () { return v1_23.cloudbilling_v1; } });
var v1_24 = require("./apis/cloudbuild/v1");
Object.defineProperty(exports, "cloudbuild_v1", { enumerable: true, get: function () { return v1_24.cloudbuild_v1; } });
var v1alpha1_2 = require("./apis/cloudbuild/v1alpha1");
Object.defineProperty(exports, "cloudbuild_v1alpha1", { enumerable: true, get: function () { return v1alpha1_2.cloudbuild_v1alpha1; } });
var v1alpha2_2 = require("./apis/cloudbuild/v1alpha2");
Object.defineProperty(exports, "cloudbuild_v1alpha2", { enumerable: true, get: function () { return v1alpha2_2.cloudbuild_v1alpha2; } });
var v2_6 = require("./apis/clouddebugger/v2");
Object.defineProperty(exports, "clouddebugger_v2", { enumerable: true, get: function () { return v2_6.clouddebugger_v2; } });
var v1beta1_10 = require("./apis/clouderrorreporting/v1beta1");
Object.defineProperty(exports, "clouderrorreporting_v1beta1", { enumerable: true, get: function () { return v1beta1_10.clouderrorreporting_v1beta1; } });
var v1_25 = require("./apis/cloudfunctions/v1");
Object.defineProperty(exports, "cloudfunctions_v1", { enumerable: true, get: function () { return v1_25.cloudfunctions_v1; } });
var v1beta2_1 = require("./apis/cloudfunctions/v1beta2");
Object.defineProperty(exports, "cloudfunctions_v1beta2", { enumerable: true, get: function () { return v1beta2_1.cloudfunctions_v1beta2; } });
var v1_26 = require("./apis/cloudidentity/v1");
Object.defineProperty(exports, "cloudidentity_v1", { enumerable: true, get: function () { return v1_26.cloudidentity_v1; } });
var v1beta1_11 = require("./apis/cloudidentity/v1beta1");
Object.defineProperty(exports, "cloudidentity_v1beta1", { enumerable: true, get: function () { return v1beta1_11.cloudidentity_v1beta1; } });
var v1_27 = require("./apis/cloudiot/v1");
Object.defineProperty(exports, "cloudiot_v1", { enumerable: true, get: function () { return v1_27.cloudiot_v1; } });
var v1_28 = require("./apis/cloudkms/v1");
Object.defineProperty(exports, "cloudkms_v1", { enumerable: true, get: function () { return v1_28.cloudkms_v1; } });
var v2_7 = require("./apis/cloudprofiler/v2");
Object.defineProperty(exports, "cloudprofiler_v2", { enumerable: true, get: function () { return v2_7.cloudprofiler_v2; } });
var v1_29 = require("./apis/cloudresourcemanager/v1");
Object.defineProperty(exports, "cloudresourcemanager_v1", { enumerable: true, get: function () { return v1_29.cloudresourcemanager_v1; } });
var v1beta1_12 = require("./apis/cloudresourcemanager/v1beta1");
Object.defineProperty(exports, "cloudresourcemanager_v1beta1", { enumerable: true, get: function () { return v1beta1_12.cloudresourcemanager_v1beta1; } });
var v2_8 = require("./apis/cloudresourcemanager/v2");
Object.defineProperty(exports, "cloudresourcemanager_v2", { enumerable: true, get: function () { return v2_8.cloudresourcemanager_v2; } });
var v2beta1_2 = require("./apis/cloudresourcemanager/v2beta1");
Object.defineProperty(exports, "cloudresourcemanager_v2beta1", { enumerable: true, get: function () { return v2beta1_2.cloudresourcemanager_v2beta1; } });
var v1_30 = require("./apis/cloudscheduler/v1");
Object.defineProperty(exports, "cloudscheduler_v1", { enumerable: true, get: function () { return v1_30.cloudscheduler_v1; } });
var v1beta1_13 = require("./apis/cloudscheduler/v1beta1");
Object.defineProperty(exports, "cloudscheduler_v1beta1", { enumerable: true, get: function () { return v1beta1_13.cloudscheduler_v1beta1; } });
var v1_31 = require("./apis/cloudsearch/v1");
Object.defineProperty(exports, "cloudsearch_v1", { enumerable: true, get: function () { return v1_31.cloudsearch_v1; } });
var v1_32 = require("./apis/cloudshell/v1");
Object.defineProperty(exports, "cloudshell_v1", { enumerable: true, get: function () { return v1_32.cloudshell_v1; } });
var v1alpha1_3 = require("./apis/cloudshell/v1alpha1");
Object.defineProperty(exports, "cloudshell_v1alpha1", { enumerable: true, get: function () { return v1alpha1_3.cloudshell_v1alpha1; } });
var v2_9 = require("./apis/cloudtasks/v2");
Object.defineProperty(exports, "cloudtasks_v2", { enumerable: true, get: function () { return v2_9.cloudtasks_v2; } });
var v2beta2_1 = require("./apis/cloudtasks/v2beta2");
Object.defineProperty(exports, "cloudtasks_v2beta2", { enumerable: true, get: function () { return v2beta2_1.cloudtasks_v2beta2; } });
var v2beta3_1 = require("./apis/cloudtasks/v2beta3");
Object.defineProperty(exports, "cloudtasks_v2beta3", { enumerable: true, get: function () { return v2beta3_1.cloudtasks_v2beta3; } });
var v1_33 = require("./apis/cloudtrace/v1");
Object.defineProperty(exports, "cloudtrace_v1", { enumerable: true, get: function () { return v1_33.cloudtrace_v1; } });
var v2_10 = require("./apis/cloudtrace/v2");
Object.defineProperty(exports, "cloudtrace_v2", { enumerable: true, get: function () { return v2_10.cloudtrace_v2; } });
var v2beta1_3 = require("./apis/cloudtrace/v2beta1");
Object.defineProperty(exports, "cloudtrace_v2beta1", { enumerable: true, get: function () { return v2beta1_3.cloudtrace_v2beta1; } });
var v1_34 = require("./apis/composer/v1");
Object.defineProperty(exports, "composer_v1", { enumerable: true, get: function () { return v1_34.composer_v1; } });
var v1beta1_14 = require("./apis/composer/v1beta1");
Object.defineProperty(exports, "composer_v1beta1", { enumerable: true, get: function () { return v1beta1_14.composer_v1beta1; } });
var alpha_1 = require("./apis/compute/alpha");
Object.defineProperty(exports, "compute_alpha", { enumerable: true, get: function () { return alpha_1.compute_alpha; } });
var beta_1 = require("./apis/compute/beta");
Object.defineProperty(exports, "compute_beta", { enumerable: true, get: function () { return beta_1.compute_beta; } });
var v1_35 = require("./apis/compute/v1");
Object.defineProperty(exports, "compute_v1", { enumerable: true, get: function () { return v1_35.compute_v1; } });
var v1_36 = require("./apis/container/v1");
Object.defineProperty(exports, "container_v1", { enumerable: true, get: function () { return v1_36.container_v1; } });
var v1beta1_15 = require("./apis/container/v1beta1");
Object.defineProperty(exports, "container_v1beta1", { enumerable: true, get: function () { return v1beta1_15.container_v1beta1; } });
var v1alpha1_4 = require("./apis/containeranalysis/v1alpha1");
Object.defineProperty(exports, "containeranalysis_v1alpha1", { enumerable: true, get: function () { return v1alpha1_4.containeranalysis_v1alpha1; } });
var v1beta1_16 = require("./apis/containeranalysis/v1beta1");
Object.defineProperty(exports, "containeranalysis_v1beta1", { enumerable: true, get: function () { return v1beta1_16.containeranalysis_v1beta1; } });
var v2_1_1 = require("./apis/content/v2.1");
Object.defineProperty(exports, "content_v2_1", { enumerable: true, get: function () { return v2_1_1.content_v2_1; } });
var v2_11 = require("./apis/content/v2");
Object.defineProperty(exports, "content_v2", { enumerable: true, get: function () { return v2_11.content_v2; } });
var v1_37 = require("./apis/customsearch/v1");
Object.defineProperty(exports, "customsearch_v1", { enumerable: true, get: function () { return v1_37.customsearch_v1; } });
var v1beta1_17 = require("./apis/datacatalog/v1beta1");
Object.defineProperty(exports, "datacatalog_v1beta1", { enumerable: true, get: function () { return v1beta1_17.datacatalog_v1beta1; } });
var v1b3_1 = require("./apis/dataflow/v1b3");
Object.defineProperty(exports, "dataflow_v1b3", { enumerable: true, get: function () { return v1b3_1.dataflow_v1b3; } });
var v1_38 = require("./apis/datafusion/v1");
Object.defineProperty(exports, "datafusion_v1", { enumerable: true, get: function () { return v1_38.datafusion_v1; } });
var v1beta1_18 = require("./apis/datafusion/v1beta1");
Object.defineProperty(exports, "datafusion_v1beta1", { enumerable: true, get: function () { return v1beta1_18.datafusion_v1beta1; } });
var v1beta1_19 = require("./apis/datalabeling/v1beta1");
Object.defineProperty(exports, "datalabeling_v1beta1", { enumerable: true, get: function () { return v1beta1_19.datalabeling_v1beta1; } });
var v1beta1_20 = require("./apis/datamigration/v1beta1");
Object.defineProperty(exports, "datamigration_v1beta1", { enumerable: true, get: function () { return v1beta1_20.datamigration_v1beta1; } });
var v1_39 = require("./apis/dataproc/v1");
Object.defineProperty(exports, "dataproc_v1", { enumerable: true, get: function () { return v1_39.dataproc_v1; } });
var v1beta2_2 = require("./apis/dataproc/v1beta2");
Object.defineProperty(exports, "dataproc_v1beta2", { enumerable: true, get: function () { return v1beta2_2.dataproc_v1beta2; } });
var v1_40 = require("./apis/datastore/v1");
Object.defineProperty(exports, "datastore_v1", { enumerable: true, get: function () { return v1_40.datastore_v1; } });
var v1beta1_21 = require("./apis/datastore/v1beta1");
Object.defineProperty(exports, "datastore_v1beta1", { enumerable: true, get: function () { return v1beta1_21.datastore_v1beta1; } });
var v1beta3_1 = require("./apis/datastore/v1beta3");
Object.defineProperty(exports, "datastore_v1beta3", { enumerable: true, get: function () { return v1beta3_1.datastore_v1beta3; } });
var alpha_2 = require("./apis/deploymentmanager/alpha");
Object.defineProperty(exports, "deploymentmanager_alpha", { enumerable: true, get: function () { return alpha_2.deploymentmanager_alpha; } });
var v2_12 = require("./apis/deploymentmanager/v2");
Object.defineProperty(exports, "deploymentmanager_v2", { enumerable: true, get: function () { return v2_12.deploymentmanager_v2; } });
var v2beta_1 = require("./apis/deploymentmanager/v2beta");
Object.defineProperty(exports, "deploymentmanager_v2beta", { enumerable: true, get: function () { return v2beta_1.deploymentmanager_v2beta; } });
var v3_3_1 = require("./apis/dfareporting/v3.3");
Object.defineProperty(exports, "dfareporting_v3_3", { enumerable: true, get: function () { return v3_3_1.dfareporting_v3_3; } });
var v3_4_1 = require("./apis/dfareporting/v3.4");
Object.defineProperty(exports, "dfareporting_v3_4", { enumerable: true, get: function () { return v3_4_1.dfareporting_v3_4; } });
var v2_13 = require("./apis/dialogflow/v2");
Object.defineProperty(exports, "dialogflow_v2", { enumerable: true, get: function () { return v2_13.dialogflow_v2; } });
var v2beta1_4 = require("./apis/dialogflow/v2beta1");
Object.defineProperty(exports, "dialogflow_v2beta1", { enumerable: true, get: function () { return v2beta1_4.dialogflow_v2beta1; } });
var v3beta1_1 = require("./apis/dialogflow/v3beta1");
Object.defineProperty(exports, "dialogflow_v3beta1", { enumerable: true, get: function () { return v3beta1_1.dialogflow_v3beta1; } });
var v1_41 = require("./apis/digitalassetlinks/v1");
Object.defineProperty(exports, "digitalassetlinks_v1", { enumerable: true, get: function () { return v1_41.digitalassetlinks_v1; } });
var v1_42 = require("./apis/discovery/v1");
Object.defineProperty(exports, "discovery_v1", { enumerable: true, get: function () { return v1_42.discovery_v1; } });
var v1_43 = require("./apis/displayvideo/v1");
Object.defineProperty(exports, "displayvideo_v1", { enumerable: true, get: function () { return v1_43.displayvideo_v1; } });
var v1beta_4 = require("./apis/displayvideo/v1beta");
Object.defineProperty(exports, "displayvideo_v1beta", { enumerable: true, get: function () { return v1beta_4.displayvideo_v1beta; } });
var v1beta2_3 = require("./apis/displayvideo/v1beta2");
Object.defineProperty(exports, "displayvideo_v1beta2", { enumerable: true, get: function () { return v1beta2_3.displayvideo_v1beta2; } });
var v1dev_1 = require("./apis/displayvideo/v1dev");
Object.defineProperty(exports, "displayvideo_v1dev", { enumerable: true, get: function () { return v1dev_1.displayvideo_v1dev; } });
var v2_14 = require("./apis/dlp/v2");
Object.defineProperty(exports, "dlp_v2", { enumerable: true, get: function () { return v2_14.dlp_v2; } });
var v1_44 = require("./apis/dns/v1");
Object.defineProperty(exports, "dns_v1", { enumerable: true, get: function () { return v1_44.dns_v1; } });
var v1beta2_4 = require("./apis/dns/v1beta2");
Object.defineProperty(exports, "dns_v1beta2", { enumerable: true, get: function () { return v1beta2_4.dns_v1beta2; } });
var v2beta1_5 = require("./apis/dns/v2beta1");
Object.defineProperty(exports, "dns_v2beta1", { enumerable: true, get: function () { return v2beta1_5.dns_v2beta1; } });
var v1_45 = require("./apis/docs/v1");
Object.defineProperty(exports, "docs_v1", { enumerable: true, get: function () { return v1_45.docs_v1; } });
var v1beta2_5 = require("./apis/documentai/v1beta2");
Object.defineProperty(exports, "documentai_v1beta2", { enumerable: true, get: function () { return v1beta2_5.documentai_v1beta2; } });
var v1beta3_2 = require("./apis/documentai/v1beta3");
Object.defineProperty(exports, "documentai_v1beta3", { enumerable: true, get: function () { return v1beta3_2.documentai_v1beta3; } });
var v1alpha2_3 = require("./apis/domains/v1alpha2");
Object.defineProperty(exports, "domains_v1alpha2", { enumerable: true, get: function () { return v1alpha2_3.domains_v1alpha2; } });
var v1beta1_22 = require("./apis/domains/v1beta1");
Object.defineProperty(exports, "domains_v1beta1", { enumerable: true, get: function () { return v1beta1_22.domains_v1beta1; } });
var v1_46 = require("./apis/domainsrdap/v1");
Object.defineProperty(exports, "domainsrdap_v1", { enumerable: true, get: function () { return v1_46.domainsrdap_v1; } });
var v1_1_2 = require("./apis/doubleclickbidmanager/v1.1");
Object.defineProperty(exports, "doubleclickbidmanager_v1_1", { enumerable: true, get: function () { return v1_1_2.doubleclickbidmanager_v1_1; } });
var v1_47 = require("./apis/doubleclickbidmanager/v1");
Object.defineProperty(exports, "doubleclickbidmanager_v1", { enumerable: true, get: function () { return v1_47.doubleclickbidmanager_v1; } });
var v2_15 = require("./apis/doubleclicksearch/v2");
Object.defineProperty(exports, "doubleclicksearch_v2", { enumerable: true, get: function () { return v2_15.doubleclicksearch_v2; } });
var v2_16 = require("./apis/drive/v2");
Object.defineProperty(exports, "drive_v2", { enumerable: true, get: function () { return v2_16.drive_v2; } });
var v3_5 = require("./apis/drive/v3");
Object.defineProperty(exports, "drive_v3", { enumerable: true, get: function () { return v3_5.drive_v3; } });
var v2_17 = require("./apis/driveactivity/v2");
Object.defineProperty(exports, "driveactivity_v2", { enumerable: true, get: function () { return v2_17.driveactivity_v2; } });
var v1beta1_23 = require("./apis/eventarc/v1beta1");
Object.defineProperty(exports, "eventarc_v1beta1", { enumerable: true, get: function () { return v1beta1_23.eventarc_v1beta1; } });
var v1alpha1_5 = require("./apis/factchecktools/v1alpha1");
Object.defineProperty(exports, "factchecktools_v1alpha1", { enumerable: true, get: function () { return v1alpha1_5.factchecktools_v1alpha1; } });
var v1_48 = require("./apis/fcm/v1");
Object.defineProperty(exports, "fcm_v1", { enumerable: true, get: function () { return v1_48.fcm_v1; } });
var v1_49 = require("./apis/file/v1");
Object.defineProperty(exports, "file_v1", { enumerable: true, get: function () { return v1_49.file_v1; } });
var v1beta1_24 = require("./apis/file/v1beta1");
Object.defineProperty(exports, "file_v1beta1", { enumerable: true, get: function () { return v1beta1_24.file_v1beta1; } });
var v1beta1_25 = require("./apis/firebase/v1beta1");
Object.defineProperty(exports, "firebase_v1beta1", { enumerable: true, get: function () { return v1beta1_25.firebase_v1beta1; } });
var v1_50 = require("./apis/firebasedynamiclinks/v1");
Object.defineProperty(exports, "firebasedynamiclinks_v1", { enumerable: true, get: function () { return v1_50.firebasedynamiclinks_v1; } });
var v1_51 = require("./apis/firebasehosting/v1");
Object.defineProperty(exports, "firebasehosting_v1", { enumerable: true, get: function () { return v1_51.firebasehosting_v1; } });
var v1beta1_26 = require("./apis/firebasehosting/v1beta1");
Object.defineProperty(exports, "firebasehosting_v1beta1", { enumerable: true, get: function () { return v1beta1_26.firebasehosting_v1beta1; } });
var v1_52 = require("./apis/firebaseml/v1");
Object.defineProperty(exports, "firebaseml_v1", { enumerable: true, get: function () { return v1_52.firebaseml_v1; } });
var v1beta2_6 = require("./apis/firebaseml/v1beta2");
Object.defineProperty(exports, "firebaseml_v1beta2", { enumerable: true, get: function () { return v1beta2_6.firebaseml_v1beta2; } });
var v1_53 = require("./apis/firebaserules/v1");
Object.defineProperty(exports, "firebaserules_v1", { enumerable: true, get: function () { return v1_53.firebaserules_v1; } });
var v1_54 = require("./apis/firestore/v1");
Object.defineProperty(exports, "firestore_v1", { enumerable: true, get: function () { return v1_54.firestore_v1; } });
var v1beta1_27 = require("./apis/firestore/v1beta1");
Object.defineProperty(exports, "firestore_v1beta1", { enumerable: true, get: function () { return v1beta1_27.firestore_v1beta1; } });
var v1beta2_7 = require("./apis/firestore/v1beta2");
Object.defineProperty(exports, "firestore_v1beta2", { enumerable: true, get: function () { return v1beta2_7.firestore_v1beta2; } });
var v1_55 = require("./apis/fitness/v1");
Object.defineProperty(exports, "fitness_v1", { enumerable: true, get: function () { return v1_55.fitness_v1; } });
var v1_56 = require("./apis/games/v1");
Object.defineProperty(exports, "games_v1", { enumerable: true, get: function () { return v1_56.games_v1; } });
var v1configuration_1 = require("./apis/gamesConfiguration/v1configuration");
Object.defineProperty(exports, "gamesConfiguration_v1configuration", { enumerable: true, get: function () { return v1configuration_1.gamesConfiguration_v1configuration; } });
var v1management_1 = require("./apis/gamesManagement/v1management");
Object.defineProperty(exports, "gamesManagement_v1management", { enumerable: true, get: function () { return v1management_1.gamesManagement_v1management; } });
var v1_57 = require("./apis/gameservices/v1");
Object.defineProperty(exports, "gameservices_v1", { enumerable: true, get: function () { return v1_57.gameservices_v1; } });
var v1beta_5 = require("./apis/gameservices/v1beta");
Object.defineProperty(exports, "gameservices_v1beta", { enumerable: true, get: function () { return v1beta_5.gameservices_v1beta; } });
var v1_58 = require("./apis/genomics/v1");
Object.defineProperty(exports, "genomics_v1", { enumerable: true, get: function () { return v1_58.genomics_v1; } });
var v1alpha2_4 = require("./apis/genomics/v1alpha2");
Object.defineProperty(exports, "genomics_v1alpha2", { enumerable: true, get: function () { return v1alpha2_4.genomics_v1alpha2; } });
var v2alpha1_1 = require("./apis/genomics/v2alpha1");
Object.defineProperty(exports, "genomics_v2alpha1", { enumerable: true, get: function () { return v2alpha1_1.genomics_v2alpha1; } });
var v1_59 = require("./apis/gmail/v1");
Object.defineProperty(exports, "gmail_v1", { enumerable: true, get: function () { return v1_59.gmail_v1; } });
var v1beta1_28 = require("./apis/gmailpostmastertools/v1beta1");
Object.defineProperty(exports, "gmailpostmastertools_v1beta1", { enumerable: true, get: function () { return v1beta1_28.gmailpostmastertools_v1beta1; } });
var v1_60 = require("./apis/groupsmigration/v1");
Object.defineProperty(exports, "groupsmigration_v1", { enumerable: true, get: function () { return v1_60.groupsmigration_v1; } });
var v1_61 = require("./apis/groupssettings/v1");
Object.defineProperty(exports, "groupssettings_v1", { enumerable: true, get: function () { return v1_61.groupssettings_v1; } });
var v1_62 = require("./apis/healthcare/v1");
Object.defineProperty(exports, "healthcare_v1", { enumerable: true, get: function () { return v1_62.healthcare_v1; } });
var v1beta1_29 = require("./apis/healthcare/v1beta1");
Object.defineProperty(exports, "healthcare_v1beta1", { enumerable: true, get: function () { return v1beta1_29.healthcare_v1beta1; } });
var v1_63 = require("./apis/homegraph/v1");
Object.defineProperty(exports, "homegraph_v1", { enumerable: true, get: function () { return v1_63.homegraph_v1; } });
var v1_64 = require("./apis/iam/v1");
Object.defineProperty(exports, "iam_v1", { enumerable: true, get: function () { return v1_64.iam_v1; } });
var v1_65 = require("./apis/iamcredentials/v1");
Object.defineProperty(exports, "iamcredentials_v1", { enumerable: true, get: function () { return v1_65.iamcredentials_v1; } });
var v1_66 = require("./apis/iap/v1");
Object.defineProperty(exports, "iap_v1", { enumerable: true, get: function () { return v1_66.iap_v1; } });
var v1beta1_30 = require("./apis/iap/v1beta1");
Object.defineProperty(exports, "iap_v1beta1", { enumerable: true, get: function () { return v1beta1_30.iap_v1beta1; } });
var v3_6 = require("./apis/identitytoolkit/v3");
Object.defineProperty(exports, "identitytoolkit_v3", { enumerable: true, get: function () { return v3_6.identitytoolkit_v3; } });
var v3_7 = require("./apis/indexing/v3");
Object.defineProperty(exports, "indexing_v3", { enumerable: true, get: function () { return v3_7.indexing_v3; } });
var v2_18 = require("./apis/jobs/v2");
Object.defineProperty(exports, "jobs_v2", { enumerable: true, get: function () { return v2_18.jobs_v2; } });
var v3_8 = require("./apis/jobs/v3");
Object.defineProperty(exports, "jobs_v3", { enumerable: true, get: function () { return v3_8.jobs_v3; } });
var v3p1beta1_1 = require("./apis/jobs/v3p1beta1");
Object.defineProperty(exports, "jobs_v3p1beta1", { enumerable: true, get: function () { return v3p1beta1_1.jobs_v3p1beta1; } });
var v1_67 = require("./apis/kgsearch/v1");
Object.defineProperty(exports, "kgsearch_v1", { enumerable: true, get: function () { return v1_67.kgsearch_v1; } });
var v1_68 = require("./apis/language/v1");
Object.defineProperty(exports, "language_v1", { enumerable: true, get: function () { return v1_68.language_v1; } });
var v1beta1_31 = require("./apis/language/v1beta1");
Object.defineProperty(exports, "language_v1beta1", { enumerable: true, get: function () { return v1beta1_31.language_v1beta1; } });
var v1beta2_8 = require("./apis/language/v1beta2");
Object.defineProperty(exports, "language_v1beta2", { enumerable: true, get: function () { return v1beta2_8.language_v1beta2; } });
var v1_69 = require("./apis/libraryagent/v1");
Object.defineProperty(exports, "libraryagent_v1", { enumerable: true, get: function () { return v1_69.libraryagent_v1; } });
var v1_70 = require("./apis/licensing/v1");
Object.defineProperty(exports, "licensing_v1", { enumerable: true, get: function () { return v1_70.licensing_v1; } });
var v2beta_2 = require("./apis/lifesciences/v2beta");
Object.defineProperty(exports, "lifesciences_v2beta", { enumerable: true, get: function () { return v2beta_2.lifesciences_v2beta; } });
var v1_71 = require("./apis/localservices/v1");
Object.defineProperty(exports, "localservices_v1", { enumerable: true, get: function () { return v1_71.localservices_v1; } });
var v2_19 = require("./apis/logging/v2");
Object.defineProperty(exports, "logging_v2", { enumerable: true, get: function () { return v2_19.logging_v2; } });
var v1_72 = require("./apis/managedidentities/v1");
Object.defineProperty(exports, "managedidentities_v1", { enumerable: true, get: function () { return v1_72.managedidentities_v1; } });
var v1alpha1_6 = require("./apis/managedidentities/v1alpha1");
Object.defineProperty(exports, "managedidentities_v1alpha1", { enumerable: true, get: function () { return v1alpha1_6.managedidentities_v1alpha1; } });
var v1beta1_32 = require("./apis/managedidentities/v1beta1");
Object.defineProperty(exports, "managedidentities_v1beta1", { enumerable: true, get: function () { return v1beta1_32.managedidentities_v1beta1; } });
var v1_73 = require("./apis/manufacturers/v1");
Object.defineProperty(exports, "manufacturers_v1", { enumerable: true, get: function () { return v1_73.manufacturers_v1; } });
var v1_74 = require("./apis/memcache/v1");
Object.defineProperty(exports, "memcache_v1", { enumerable: true, get: function () { return v1_74.memcache_v1; } });
var v1beta2_9 = require("./apis/memcache/v1beta2");
Object.defineProperty(exports, "memcache_v1beta2", { enumerable: true, get: function () { return v1beta2_9.memcache_v1beta2; } });
var v1_75 = require("./apis/ml/v1");
Object.defineProperty(exports, "ml_v1", { enumerable: true, get: function () { return v1_75.ml_v1; } });
var v1_76 = require("./apis/monitoring/v1");
Object.defineProperty(exports, "monitoring_v1", { enumerable: true, get: function () { return v1_76.monitoring_v1; } });
var v3_9 = require("./apis/monitoring/v3");
Object.defineProperty(exports, "monitoring_v3", { enumerable: true, get: function () { return v3_9.monitoring_v3; } });
var v1_77 = require("./apis/networkmanagement/v1");
Object.defineProperty(exports, "networkmanagement_v1", { enumerable: true, get: function () { return v1_77.networkmanagement_v1; } });
var v1beta1_33 = require("./apis/networkmanagement/v1beta1");
Object.defineProperty(exports, "networkmanagement_v1beta1", { enumerable: true, get: function () { return v1beta1_33.networkmanagement_v1beta1; } });
var v2_20 = require("./apis/oauth2/v2");
Object.defineProperty(exports, "oauth2_v2", { enumerable: true, get: function () { return v2_20.oauth2_v2; } });
var v1_78 = require("./apis/osconfig/v1");
Object.defineProperty(exports, "osconfig_v1", { enumerable: true, get: function () { return v1_78.osconfig_v1; } });
var v1beta_6 = require("./apis/osconfig/v1beta");
Object.defineProperty(exports, "osconfig_v1beta", { enumerable: true, get: function () { return v1beta_6.osconfig_v1beta; } });
var v1_79 = require("./apis/oslogin/v1");
Object.defineProperty(exports, "oslogin_v1", { enumerable: true, get: function () { return v1_79.oslogin_v1; } });
var v1alpha_4 = require("./apis/oslogin/v1alpha");
Object.defineProperty(exports, "oslogin_v1alpha", { enumerable: true, get: function () { return v1alpha_4.oslogin_v1alpha; } });
var v1beta_7 = require("./apis/oslogin/v1beta");
Object.defineProperty(exports, "oslogin_v1beta", { enumerable: true, get: function () { return v1beta_7.oslogin_v1beta; } });
var v5_1 = require("./apis/pagespeedonline/v5");
Object.defineProperty(exports, "pagespeedonline_v5", { enumerable: true, get: function () { return v5_1.pagespeedonline_v5; } });
var v1_80 = require("./apis/people/v1");
Object.defineProperty(exports, "people_v1", { enumerable: true, get: function () { return v1_80.people_v1; } });
var v3_10 = require("./apis/playablelocations/v3");
Object.defineProperty(exports, "playablelocations_v3", { enumerable: true, get: function () { return v3_10.playablelocations_v3; } });
var v1_81 = require("./apis/playcustomapp/v1");
Object.defineProperty(exports, "playcustomapp_v1", { enumerable: true, get: function () { return v1_81.playcustomapp_v1; } });
var v1_82 = require("./apis/plus/v1");
Object.defineProperty(exports, "plus_v1", { enumerable: true, get: function () { return v1_82.plus_v1; } });
var v1_83 = require("./apis/policytroubleshooter/v1");
Object.defineProperty(exports, "policytroubleshooter_v1", { enumerable: true, get: function () { return v1_83.policytroubleshooter_v1; } });
var v1beta_8 = require("./apis/policytroubleshooter/v1beta");
Object.defineProperty(exports, "policytroubleshooter_v1beta", { enumerable: true, get: function () { return v1beta_8.policytroubleshooter_v1beta; } });
var v1_84 = require("./apis/poly/v1");
Object.defineProperty(exports, "poly_v1", { enumerable: true, get: function () { return v1_84.poly_v1; } });
var v1beta1_34 = require("./apis/privateca/v1beta1");
Object.defineProperty(exports, "privateca_v1beta1", { enumerable: true, get: function () { return v1beta1_34.privateca_v1beta1; } });
var v1alpha1_7 = require("./apis/prod_tt_sasportal/v1alpha1");
Object.defineProperty(exports, "prod_tt_sasportal_v1alpha1", { enumerable: true, get: function () { return v1alpha1_7.prod_tt_sasportal_v1alpha1; } });
var v1_85 = require("./apis/pubsub/v1");
Object.defineProperty(exports, "pubsub_v1", { enumerable: true, get: function () { return v1_85.pubsub_v1; } });
var v1beta1a_1 = require("./apis/pubsub/v1beta1a");
Object.defineProperty(exports, "pubsub_v1beta1a", { enumerable: true, get: function () { return v1beta1a_1.pubsub_v1beta1a; } });
var v1beta2_10 = require("./apis/pubsub/v1beta2");
Object.defineProperty(exports, "pubsub_v1beta2", { enumerable: true, get: function () { return v1beta2_10.pubsub_v1beta2; } });
var v1_86 = require("./apis/pubsublite/v1");
Object.defineProperty(exports, "pubsublite_v1", { enumerable: true, get: function () { return v1_86.pubsublite_v1; } });
var v1_87 = require("./apis/realtimebidding/v1");
Object.defineProperty(exports, "realtimebidding_v1", { enumerable: true, get: function () { return v1_87.realtimebidding_v1; } });
var v1beta1_35 = require("./apis/recommendationengine/v1beta1");
Object.defineProperty(exports, "recommendationengine_v1beta1", { enumerable: true, get: function () { return v1beta1_35.recommendationengine_v1beta1; } });
var v1_88 = require("./apis/recommender/v1");
Object.defineProperty(exports, "recommender_v1", { enumerable: true, get: function () { return v1_88.recommender_v1; } });
var v1beta1_36 = require("./apis/recommender/v1beta1");
Object.defineProperty(exports, "recommender_v1beta1", { enumerable: true, get: function () { return v1beta1_36.recommender_v1beta1; } });
var v1_89 = require("./apis/redis/v1");
Object.defineProperty(exports, "redis_v1", { enumerable: true, get: function () { return v1_89.redis_v1; } });
var v1beta1_37 = require("./apis/redis/v1beta1");
Object.defineProperty(exports, "redis_v1beta1", { enumerable: true, get: function () { return v1beta1_37.redis_v1beta1; } });
var v1_90 = require("./apis/remotebuildexecution/v1");
Object.defineProperty(exports, "remotebuildexecution_v1", { enumerable: true, get: function () { return v1_90.remotebuildexecution_v1; } });
var v1alpha_5 = require("./apis/remotebuildexecution/v1alpha");
Object.defineProperty(exports, "remotebuildexecution_v1alpha", { enumerable: true, get: function () { return v1alpha_5.remotebuildexecution_v1alpha; } });
var v2_21 = require("./apis/remotebuildexecution/v2");
Object.defineProperty(exports, "remotebuildexecution_v2", { enumerable: true, get: function () { return v2_21.remotebuildexecution_v2; } });
var v1_91 = require("./apis/reseller/v1");
Object.defineProperty(exports, "reseller_v1", { enumerable: true, get: function () { return v1_91.reseller_v1; } });
var v1_92 = require("./apis/run/v1");
Object.defineProperty(exports, "run_v1", { enumerable: true, get: function () { return v1_92.run_v1; } });
var v1alpha1_8 = require("./apis/run/v1alpha1");
Object.defineProperty(exports, "run_v1alpha1", { enumerable: true, get: function () { return v1alpha1_8.run_v1alpha1; } });
var v1beta1_38 = require("./apis/run/v1beta1");
Object.defineProperty(exports, "run_v1beta1", { enumerable: true, get: function () { return v1beta1_38.run_v1beta1; } });
var v1_93 = require("./apis/runtimeconfig/v1");
Object.defineProperty(exports, "runtimeconfig_v1", { enumerable: true, get: function () { return v1_93.runtimeconfig_v1; } });
var v1beta1_39 = require("./apis/runtimeconfig/v1beta1");
Object.defineProperty(exports, "runtimeconfig_v1beta1", { enumerable: true, get: function () { return v1beta1_39.runtimeconfig_v1beta1; } });
var v4_2 = require("./apis/safebrowsing/v4");
Object.defineProperty(exports, "safebrowsing_v4", { enumerable: true, get: function () { return v4_2.safebrowsing_v4; } });
var v1alpha1_9 = require("./apis/sasportal/v1alpha1");
Object.defineProperty(exports, "sasportal_v1alpha1", { enumerable: true, get: function () { return v1alpha1_9.sasportal_v1alpha1; } });
var v1_94 = require("./apis/script/v1");
Object.defineProperty(exports, "script_v1", { enumerable: true, get: function () { return v1_94.script_v1; } });
var v1_95 = require("./apis/searchconsole/v1");
Object.defineProperty(exports, "searchconsole_v1", { enumerable: true, get: function () { return v1_95.searchconsole_v1; } });
var v1_96 = require("./apis/secretmanager/v1");
Object.defineProperty(exports, "secretmanager_v1", { enumerable: true, get: function () { return v1_96.secretmanager_v1; } });
var v1beta1_40 = require("./apis/secretmanager/v1beta1");
Object.defineProperty(exports, "secretmanager_v1beta1", { enumerable: true, get: function () { return v1beta1_40.secretmanager_v1beta1; } });
var v1_97 = require("./apis/securitycenter/v1");
Object.defineProperty(exports, "securitycenter_v1", { enumerable: true, get: function () { return v1_97.securitycenter_v1; } });
var v1beta1_41 = require("./apis/securitycenter/v1beta1");
Object.defineProperty(exports, "securitycenter_v1beta1", { enumerable: true, get: function () { return v1beta1_41.securitycenter_v1beta1; } });
var v1beta2_11 = require("./apis/securitycenter/v1beta2");
Object.defineProperty(exports, "securitycenter_v1beta2", { enumerable: true, get: function () { return v1beta2_11.securitycenter_v1beta2; } });
var v1p1alpha1_1 = require("./apis/securitycenter/v1p1alpha1");
Object.defineProperty(exports, "securitycenter_v1p1alpha1", { enumerable: true, get: function () { return v1p1alpha1_1.securitycenter_v1p1alpha1; } });
var v1p1beta1_2 = require("./apis/securitycenter/v1p1beta1");
Object.defineProperty(exports, "securitycenter_v1p1beta1", { enumerable: true, get: function () { return v1p1beta1_2.securitycenter_v1p1beta1; } });
var v1_98 = require("./apis/serviceconsumermanagement/v1");
Object.defineProperty(exports, "serviceconsumermanagement_v1", { enumerable: true, get: function () { return v1_98.serviceconsumermanagement_v1; } });
var v1beta1_42 = require("./apis/serviceconsumermanagement/v1beta1");
Object.defineProperty(exports, "serviceconsumermanagement_v1beta1", { enumerable: true, get: function () { return v1beta1_42.serviceconsumermanagement_v1beta1; } });
var v1_99 = require("./apis/servicecontrol/v1");
Object.defineProperty(exports, "servicecontrol_v1", { enumerable: true, get: function () { return v1_99.servicecontrol_v1; } });
var v2_22 = require("./apis/servicecontrol/v2");
Object.defineProperty(exports, "servicecontrol_v2", { enumerable: true, get: function () { return v2_22.servicecontrol_v2; } });
var v1beta1_43 = require("./apis/servicedirectory/v1beta1");
Object.defineProperty(exports, "servicedirectory_v1beta1", { enumerable: true, get: function () { return v1beta1_43.servicedirectory_v1beta1; } });
var v1_100 = require("./apis/servicemanagement/v1");
Object.defineProperty(exports, "servicemanagement_v1", { enumerable: true, get: function () { return v1_100.servicemanagement_v1; } });
var v1_101 = require("./apis/servicenetworking/v1");
Object.defineProperty(exports, "servicenetworking_v1", { enumerable: true, get: function () { return v1_101.servicenetworking_v1; } });
var v1beta_9 = require("./apis/servicenetworking/v1beta");
Object.defineProperty(exports, "servicenetworking_v1beta", { enumerable: true, get: function () { return v1beta_9.servicenetworking_v1beta; } });
var v1_102 = require("./apis/serviceusage/v1");
Object.defineProperty(exports, "serviceusage_v1", { enumerable: true, get: function () { return v1_102.serviceusage_v1; } });
var v1beta1_44 = require("./apis/serviceusage/v1beta1");
Object.defineProperty(exports, "serviceusage_v1beta1", { enumerable: true, get: function () { return v1beta1_44.serviceusage_v1beta1; } });
var v4_3 = require("./apis/sheets/v4");
Object.defineProperty(exports, "sheets_v4", { enumerable: true, get: function () { return v4_3.sheets_v4; } });
var v1_103 = require("./apis/siteVerification/v1");
Object.defineProperty(exports, "siteVerification_v1", { enumerable: true, get: function () { return v1_103.siteVerification_v1; } });
var v1_104 = require("./apis/slides/v1");
Object.defineProperty(exports, "slides_v1", { enumerable: true, get: function () { return v1_104.slides_v1; } });
var v1_105 = require("./apis/smartdevicemanagement/v1");
Object.defineProperty(exports, "smartdevicemanagement_v1", { enumerable: true, get: function () { return v1_105.smartdevicemanagement_v1; } });
var v1_106 = require("./apis/sourcerepo/v1");
Object.defineProperty(exports, "sourcerepo_v1", { enumerable: true, get: function () { return v1_106.sourcerepo_v1; } });
var v1_107 = require("./apis/spanner/v1");
Object.defineProperty(exports, "spanner_v1", { enumerable: true, get: function () { return v1_107.spanner_v1; } });
var v1_108 = require("./apis/speech/v1");
Object.defineProperty(exports, "speech_v1", { enumerable: true, get: function () { return v1_108.speech_v1; } });
var v1p1beta1_3 = require("./apis/speech/v1p1beta1");
Object.defineProperty(exports, "speech_v1p1beta1", { enumerable: true, get: function () { return v1p1beta1_3.speech_v1p1beta1; } });
var v2beta1_6 = require("./apis/speech/v2beta1");
Object.defineProperty(exports, "speech_v2beta1", { enumerable: true, get: function () { return v2beta1_6.speech_v2beta1; } });
var v1beta4_1 = require("./apis/sql/v1beta4");
Object.defineProperty(exports, "sql_v1beta4", { enumerable: true, get: function () { return v1beta4_1.sql_v1beta4; } });
var v1beta4_2 = require("./apis/sqladmin/v1beta4");
Object.defineProperty(exports, "sqladmin_v1beta4", { enumerable: true, get: function () { return v1beta4_2.sqladmin_v1beta4; } });
var v1_109 = require("./apis/storage/v1");
Object.defineProperty(exports, "storage_v1", { enumerable: true, get: function () { return v1_109.storage_v1; } });
var v1beta2_12 = require("./apis/storage/v1beta2");
Object.defineProperty(exports, "storage_v1beta2", { enumerable: true, get: function () { return v1beta2_12.storage_v1beta2; } });
var v1_110 = require("./apis/storagetransfer/v1");
Object.defineProperty(exports, "storagetransfer_v1", { enumerable: true, get: function () { return v1_110.storagetransfer_v1; } });
var v1_111 = require("./apis/streetviewpublish/v1");
Object.defineProperty(exports, "streetviewpublish_v1", { enumerable: true, get: function () { return v1_111.streetviewpublish_v1; } });
var v1_112 = require("./apis/sts/v1");
Object.defineProperty(exports, "sts_v1", { enumerable: true, get: function () { return v1_112.sts_v1; } });
var v1beta_10 = require("./apis/sts/v1beta");
Object.defineProperty(exports, "sts_v1beta", { enumerable: true, get: function () { return v1beta_10.sts_v1beta; } });
var v1_113 = require("./apis/tagmanager/v1");
Object.defineProperty(exports, "tagmanager_v1", { enumerable: true, get: function () { return v1_113.tagmanager_v1; } });
var v2_23 = require("./apis/tagmanager/v2");
Object.defineProperty(exports, "tagmanager_v2", { enumerable: true, get: function () { return v2_23.tagmanager_v2; } });
var v1_114 = require("./apis/tasks/v1");
Object.defineProperty(exports, "tasks_v1", { enumerable: true, get: function () { return v1_114.tasks_v1; } });
var v1_115 = require("./apis/testing/v1");
Object.defineProperty(exports, "testing_v1", { enumerable: true, get: function () { return v1_115.testing_v1; } });
var v1_116 = require("./apis/texttospeech/v1");
Object.defineProperty(exports, "texttospeech_v1", { enumerable: true, get: function () { return v1_116.texttospeech_v1; } });
var v1beta1_45 = require("./apis/texttospeech/v1beta1");
Object.defineProperty(exports, "texttospeech_v1beta1", { enumerable: true, get: function () { return v1beta1_45.texttospeech_v1beta1; } });
var v1beta3_3 = require("./apis/toolresults/v1beta3");
Object.defineProperty(exports, "toolresults_v1beta3", { enumerable: true, get: function () { return v1beta3_3.toolresults_v1beta3; } });
var v1_117 = require("./apis/tpu/v1");
Object.defineProperty(exports, "tpu_v1", { enumerable: true, get: function () { return v1_117.tpu_v1; } });
var v1alpha1_10 = require("./apis/tpu/v1alpha1");
Object.defineProperty(exports, "tpu_v1alpha1", { enumerable: true, get: function () { return v1alpha1_10.tpu_v1alpha1; } });
var v2_24 = require("./apis/trafficdirector/v2");
Object.defineProperty(exports, "trafficdirector_v2", { enumerable: true, get: function () { return v2_24.trafficdirector_v2; } });
var v2_25 = require("./apis/translate/v2");
Object.defineProperty(exports, "translate_v2", { enumerable: true, get: function () { return v2_25.translate_v2; } });
var v3_11 = require("./apis/translate/v3");
Object.defineProperty(exports, "translate_v3", { enumerable: true, get: function () { return v3_11.translate_v3; } });
var v3beta1_2 = require("./apis/translate/v3beta1");
Object.defineProperty(exports, "translate_v3beta1", { enumerable: true, get: function () { return v3beta1_2.translate_v3beta1; } });
var v1_118 = require("./apis/vault/v1");
Object.defineProperty(exports, "vault_v1", { enumerable: true, get: function () { return v1_118.vault_v1; } });
var v1_119 = require("./apis/vectortile/v1");
Object.defineProperty(exports, "vectortile_v1", { enumerable: true, get: function () { return v1_119.vectortile_v1; } });
var v1_120 = require("./apis/verifiedaccess/v1");
Object.defineProperty(exports, "verifiedaccess_v1", { enumerable: true, get: function () { return v1_120.verifiedaccess_v1; } });
var v1_121 = require("./apis/videointelligence/v1");
Object.defineProperty(exports, "videointelligence_v1", { enumerable: true, get: function () { return v1_121.videointelligence_v1; } });
var v1beta2_13 = require("./apis/videointelligence/v1beta2");
Object.defineProperty(exports, "videointelligence_v1beta2", { enumerable: true, get: function () { return v1beta2_13.videointelligence_v1beta2; } });
var v1p1beta1_4 = require("./apis/videointelligence/v1p1beta1");
Object.defineProperty(exports, "videointelligence_v1p1beta1", { enumerable: true, get: function () { return v1p1beta1_4.videointelligence_v1p1beta1; } });
var v1p2beta1_1 = require("./apis/videointelligence/v1p2beta1");
Object.defineProperty(exports, "videointelligence_v1p2beta1", { enumerable: true, get: function () { return v1p2beta1_1.videointelligence_v1p2beta1; } });
var v1p3beta1_1 = require("./apis/videointelligence/v1p3beta1");
Object.defineProperty(exports, "videointelligence_v1p3beta1", { enumerable: true, get: function () { return v1p3beta1_1.videointelligence_v1p3beta1; } });
var v1_122 = require("./apis/vision/v1");
Object.defineProperty(exports, "vision_v1", { enumerable: true, get: function () { return v1_122.vision_v1; } });
var v1p1beta1_5 = require("./apis/vision/v1p1beta1");
Object.defineProperty(exports, "vision_v1p1beta1", { enumerable: true, get: function () { return v1p1beta1_5.vision_v1p1beta1; } });
var v1p2beta1_2 = require("./apis/vision/v1p2beta1");
Object.defineProperty(exports, "vision_v1p2beta1", { enumerable: true, get: function () { return v1p2beta1_2.vision_v1p2beta1; } });
var v1_123 = require("./apis/webfonts/v1");
Object.defineProperty(exports, "webfonts_v1", { enumerable: true, get: function () { return v1_123.webfonts_v1; } });
var v3_12 = require("./apis/webmasters/v3");
Object.defineProperty(exports, "webmasters_v3", { enumerable: true, get: function () { return v3_12.webmasters_v3; } });
var v1_124 = require("./apis/websecurityscanner/v1");
Object.defineProperty(exports, "websecurityscanner_v1", { enumerable: true, get: function () { return v1_124.websecurityscanner_v1; } });
var v1alpha_6 = require("./apis/websecurityscanner/v1alpha");
Object.defineProperty(exports, "websecurityscanner_v1alpha", { enumerable: true, get: function () { return v1alpha_6.websecurityscanner_v1alpha; } });
var v1beta_11 = require("./apis/websecurityscanner/v1beta");
Object.defineProperty(exports, "websecurityscanner_v1beta", { enumerable: true, get: function () { return v1beta_11.websecurityscanner_v1beta; } });
var v1beta_12 = require("./apis/workflowexecutions/v1beta");
Object.defineProperty(exports, "workflowexecutions_v1beta", { enumerable: true, get: function () { return v1beta_12.workflowexecutions_v1beta; } });
var v1beta_13 = require("./apis/workflows/v1beta");
Object.defineProperty(exports, "workflows_v1beta", { enumerable: true, get: function () { return v1beta_13.workflows_v1beta; } });
var v3_13 = require("./apis/youtube/v3");
Object.defineProperty(exports, "youtube_v3", { enumerable: true, get: function () { return v3_13.youtube_v3; } });
var v1_125 = require("./apis/youtubeAnalytics/v1");
Object.defineProperty(exports, "youtubeAnalytics_v1", { enumerable: true, get: function () { return v1_125.youtubeAnalytics_v1; } });
var v2_26 = require("./apis/youtubeAnalytics/v2");
Object.defineProperty(exports, "youtubeAnalytics_v2", { enumerable: true, get: function () { return v2_26.youtubeAnalytics_v2; } });
var v1_126 = require("./apis/youtubereporting/v1");
Object.defineProperty(exports, "youtubereporting_v1", { enumerable: true, get: function () { return v1_126.youtubereporting_v1; } });
