// googleapis/build/src/apis/speech/index.js
"use strict";
// Copyright 2019 Google LLC
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
/*! THIS FILE IS AUTO-GENERATED */
const googleapis_common_1 = require("googleapis-common");
const v1_1 = require("./v1");
const v1p1beta1_1 = require("./v1p1beta1");
const v2beta1_1 = require("./v2beta1");
exports.VERSIONS = {
    v1: v1_1.speech_v1.Speech,
    v1p1beta1: v1p1beta1_1.speech_v1p1beta1.Speech,
    v2beta1: v2beta1_1.speech_v2beta1.Speech,
};
function speech(versionOrOptions) {
    return googleapis_common_1.getAPI('speech', versionOrOptions, exports.VERSIONS, this);
}
exports.speech = speech;
const auth = new googleapis_common_1.AuthPlus();
exports.auth = auth;
